### ISSUE TITLE
Bug: referencing invalid `config` keys causes silent exit

### ISSUE THREAD

**Issue by osterman** [mumoshu/variant2#7]
If I'm referencing my configuration loaded via `config { ...}` and the key does not exist, then `variant` just exits with no error message.

**Comment by mumoshu** [comment]
@osterman Thanks. Would you mind sharing me an example for reproduction? It will help me fix the issue so much faster!

**Comment by osterman** [comment]
Will do... here's one related, but different. This one causes a segfault.

```
#!/usr/bin/env variant

config "settings" {
  source file {
    path = "${context.sourcedir}/conf/test.yaml"
    default = ""
  }
}

job "example" {
  description = "Write varfile to terraform moduledir"
  variable "test" {
    value = "123"
  }
  exec {
      command = "echo"
      args = [
        "[${conf.settings.projects[var.test]}]"
      ]
   }
}
```

with a file in `conf/test.yml`:

```
projects:
   something: here
```

**Comment by osterman** [comment]
I'm not able to reproduce the original error. Not sure why. 

I am getting a *good* error:
> Unsupported attribute; This object does not have an attribute named "test"., and 1 other diagnostic(s)

**Comment by mumoshu** [comment]
@osterman The panic seems to have happened due to missing `type` in the `variable` block. That's fixed via 
2072059.

I've also added some error printing logics in a few possible places to hopefully fix the original issue fc5484f

Please reopen or submit another issue(s) if you managed to reproduce it.
